app/javascript/onboarding/components/EmailListTermsConditionsForm.jsx

```diff
@@ -13,10 +13,10 @@ class EmailTermsConditionsForm extends Component {
     this.checkRequirements = this.checkRequirements.bind(this);
 
     this.state = {
-      checked_code_of_conduct: false,
-      checked_terms_and_conditions: false,
-      email_membership_newsletter: true,
-      email_digest_periodic: true,
+      checkedCodeOfConduct: false,
+      checkedTermsAndConditions: false,
+      emailMembershipNewsletter: true,
+      emailDigestPeriodic: true,
```

Comment:
Not that this will technically be breaking the onboard process, but it won't properly update user's preference. This is because the [whole state is sent via fetch](https://github.com/thepracticaldev/dev.to/blob/16e17843ed85e3095b12437fc2c9e41be05770d4/app/javascript/onboarding/components/EmailListTermsConditionsForm.jsx#L50) as param.

I still want to accept this and make changes to the [controller](https://github.com/thepracticaldev/dev.to/blob/master/app/controllers/users_controller.rb#L114) to snake-case params but reverting this refactor could be more straight foward. Do you have any thought on this @jacobherrington ?

Reply:
Firstly, we probably want a feature spec to validate that onboarding works, going forward. That's something I'm happy to tackle in another PR.

Secondly, I think we should revert these changes in this PR and handle that in a new PR; I believe there are quite a few ways to address this and I'm not sure the Rails community has settled on one. Having it in a separate PR will allow us to merge this one and discuss it at length.

It's definitely a little weird to see snake case in JavaScript, but it's also weird to see camel case in Ruby.

I found [an answer to this on SO](https://stackoverflow.com/a/30557924/6010917), but I'm not sure it's the best route.